### PR TITLE
[Merged by Bors] - feat: port Init/Algebra/Classes.lean

### PR DESCRIPTION
**Pull request by kim-em**

### CODE REVIEW

File: Mathlib/Init/Algebra/Classes.lean [hunk 1]
```diff
@@ -0,0 +1,447 @@
...
+Copyright (c) 2017 Microsoft Corporation. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Leonardo de Moura
+-/
+import Mathlib.Init.Logic
+
+/-!
+# Unbundled algebra classes
+
+These classes and the `@[algebra]` attribute are part of an incomplete refactor described
```
Comment (PatrickMassot) [inline Mathlib/Init/Algebra/Classes.lean:11]: At least the part about `@[algebra]` should be dropped, right?

File: Mathlib/Init/Algebra/Classes.lean [hunk 2]
```diff
@@ -0,0 +1,447 @@
...
+* `is_per`: unused
+* `is_incomp_trans`: unused
+* `is_strict_weak_order`: significant usage (most of it on `rbmap`, could be transferred)
+* `is_trichotomous`: some usage
+* `is_strict_total_order`: looks like the only usage is in `rbmap` again
+-/
+
+universe u v
+
+-- @[algebra]
```
Comment (PatrickMassot) [inline Mathlib/Init/Algebra/Classes.lean:63]: Is there any point in keeping those comments? Is this attribute doing anything in Lean 3?

Comment (digama0) [reply Mathlib/Init/Algebra/Classes.lean:63]: Mathport knows how to drop attributes, I think it's safe to drop this one since it never did anything.

**Comment by PatrickMassot** [comment]
bors merge

**Pull request closed without merging**